web/permissions/permissions.go

```diff
@@ -36,6 +41,11 @@ func keyPicker(i *instance.Instance) jwt.Keyfunc {
 		case permissions.AppAudience:
 			return i.SessionSecret, nil
 		case permissions.RefreshTokenAudience, permissions.AccessTokenAudience:
+			// An OAuth2 token is only valid if the client has not been revoked
```

Comment:
I feel this should probably be in extractJWTClaims : 
- keyPicker responsibility is to choose the key (it handle rotation for instance)
- extractJWTClaims does the token validation (domain, ...)